modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -405,8 +407,25 @@ private Point2D convertSwingToFxPixel(GraphicsConfiguration g, double wx, double
         return new Point2D(newx, newy);
     }
 
+    private EmbeddedSceneInterface getScenePeer() {
+        EmbeddedSceneInterface lScenePeer;
+        synchronized(LOCK) {
+            lScenePeer = scenePeer;
+        }
+        return lScenePeer;
+    }
+
+    private EmbeddedStageInterface getStagePeer() {
+        EmbeddedStageInterface lStagePeer;
+        synchronized(LOCK) {
+            lStagePeer = stagePeer;
+        }
+        return lStagePeer;
+    }
+
     private void sendMouseEventToFX(MouseEvent e) {
-        if (scenePeer == null || !isFxEnabled()) {
+        EmbeddedSceneInterface lScenePeer = getScenePeer();
+        if (lScenePeer == null || !isFxEnabled()) {
```

Comment:
Probably minor, but maybe we should first check `!isFxEnabled()` and return fast before we actually call `getScenePeer()` and thus locking it

Reply:
ok..sure..